hoss/coordinate_utilities.py

```diff
@@ -179,42 +184,225 @@ def get_1D_dim_array_data_from_dimvalues(
+        from_geo_transformer.transform(point_latitudes, point_longitudes)
+    )
+    x_y_points = list(zip(points_x, points_y))
+
+    return x_y_points
+
+
+def get_x_y_values_from_geographic_points1(
+    points: Dict, crs: CRS
+) -> Dict[tuple, tuple]:
+    """Take an input list of (longitude, latitude) coordinates and project
+    those points to the target grid. Then return the x-y dimscales
+
+    """
+
+    point_longitudes, point_latitudes = zip(*list(points.values()))
+
+    from_geo_transformer = Transformer.from_crs(4326, crs)
+    points_x, points_y = (  # pylint: disable=unpacking-non-sequence
+        from_geo_transformer.transform(point_latitudes, point_longitudes)
+    )
+
+    x_y_points = {}
+    for index, point_x, point_y in zip(list(points.keys()), points_x, points_y):
+        x_y_points.update({index: (point_x, point_y)})
+
+    return x_y_points
+
+
+def get_row_col_valid_indices_in_dataset(
+    lat_arr: np.ndarray,
+    lon_arr: np.ndarray,
+    lat_coordinate: VariableFromDmr,
+    lon_coordinate: VariableFromDmr,
+) -> tuple[list, list]:
+    """
+    This function gets valid indices across rows and columns of
+    both the latitude and longitude datasets
+    """
+    valid_lat_lon_mask = np.logical_and(
+        get_valid_indices(lat_arr, lat_coordinate),
+        get_valid_indices(lon_arr, lon_coordinate),
+    )
+
+    # get maximally spread points within rows
+    max_x_spread_pts = get_max_x_spread_pts(~valid_lat_lon_mask)
+
+    # Doing the same for the columns is done by transposing the valid_mask
+    # and then fixing the results from [x, y] to [y, x].
+    max_y_spread_trsp = get_max_x_spread_pts(np.transpose(~valid_lat_lon_mask))
+    max_y_spread_pts = [
+        np.flip(max_y_spread_trsp[0]),
+        np.flip(max_y_spread_trsp[1]),
+    ]
+
+    return max_y_spread_pts, max_x_spread_pts
+
+
+def get_max_x_spread_pts(
+    valid_mask: np.ndarray,  # Numpy Mask Array, e.g., invalid latitudes & longitudes
+) -> list[list]:  # 2 points by indices, [[y_ind, x_ind], [y_ind, x_ind]
+    """
+    # This function returns two data points by x, y indices that are spread farthest
+    # from each other in the same row, i.e., have the greatest delta-x value - and
+    # are valid data points from the valid_mask array passed in. The input array
+    # must be a 2D Numpy mask array providing the valid data points, e.g., filtering
+    # out fill values and out-of-range values.
+    """
+    # fill a sample array with x-index values, x_ind[i, j] = j
+    x_ind = np.array(
+        [
+            # [j for j in range(valid_mask.shape[1])]
+            list(range(valid_mask.shape[1]))
+            for i in range(valid_mask.shape[0])
+        ]
+    )
+    # mask x_ind to hide the invalid data points
+    valid_x_ind = np.ma.array(x_ind, mask=valid_mask)
+
+    # ptp (peak-to-peak) finds the greatest delta-x value amongst valid points
+    # for each row. Result is 1D
+    x_ind_spread = valid_x_ind.ptp(axis=1)
```

Comment:
If there is only one valid value, I think this gives you a single result (of value 0). It feels like that needs to be caught here before proceeding. (I don't _think_ right now you are capturing the case anywhere that there is only a single valid point, apologies if I missed that, though)

Reply:
yeah. Good catch. I think I had that check in the previous version of the method...
Will add the check

Reply:
updated - https://github.com/nasa/harmony-opendap-subsetter/pull/18/commits/2f07d1a475b2033b4ee78de17d353d22729addc1